lib/tapioca/dsl/compilers/job_iteration.rb

```diff
@@ -0,0 +1,96 @@
+# typed: strict
+# frozen_string_literal: true
+
+return unless defined?(JobIteration::Iteration)
+
+require "prism"
+
+module Tapioca
+  module Dsl
+    module Compilers
+      class JobIteration < Compiler
+        extend T::Sig
+
+        ConstantType = type_member { { fixed: T.class_of(::JobIteration::Iteration) } }
+
+        sig { override.void }
+        def decorate
+          return unless constant.instance_methods(false).include?(:build_enumerator)
+
+          root.create_path(constant) do |job|
+            method = constant.instance_method(:build_enumerator)
+            constant_name = name_of(constant)
+            parameters = compile_method_parameters_to_rbi(method).reject do |typed_param|
+              typed_param.param.name == "cursor"
+            end
+
+            expanded_parameters = parameters.flat_map do |typed_param|
+              hash_param = typed_param.type.match(/\A\{.*\}\z/)
+              if hash_param
+                key_value_pairs = parse_hash_parameter(typed_param)
+                key_value_pairs.map do |key, value|
+                  create_kw_param(key, type: value)
+                end
+              else
+                typed_param
+              end
+            end
+
+            return_type = compile_method_return_type_to_rbi(method)
+
+            job.create_method(
+              "perform_later",
+              parameters: expanded_parameters,
```

Comment:
We're missing the potential block here: https://github.com/Shopify/tapioca/blob/91ad27251ad8c422f116cd0eafe9a5bd82ea7d70/lib/tapioca/dsl/compilers/active_job.rb#L32